packages/realm/src/app-services/utils.ts

```diff
@@ -16,11 +16,15 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+import { binding } from "src/internal";
+
 /**
  * Remove entries for undefined property values.
  * @internal
  */
-export function cleanArguments(args: unknown[] | unknown): unknown[] | unknown {
+export function cleanArguments(args: unknown[]): binding.EJson[];
+export function cleanArguments(args: unknown): binding.EJson;
+export function cleanArguments(args: unknown[] | unknown) {
```

Comment:
While I was at it, I updated the signature of this, to avoid type assertions on the calling side.